cleo_sdk/CLEO_Utils.h

```diff
@@ -197,16 +197,20 @@ namespace CLEO
         if (!std::filesystem::path(path).is_absolute())
         {
             absolute = CLEO_GetScriptWorkDir(thread);
-            absolute += '\\';
-            absolute += path;
-            FilepathNormalize(absolute, false);
-            path = absolute.c_str();
+            if (!absolute.empty())
+            {
+                absolute += '\\';
+                absolute += path;
+                FilepathNormalize(absolute, false);
+                path = absolute.c_str();
+            }
         }
 
         if (!StringStartsWith(path, CLEO_GetGameDirectory(), false) &&
             !StringStartsWith(path, CLEO_GetUserDirectory(), false))
         {
-            return false;
+            if (StringStartsWith(path, "..")) // relative path trying to escape game's directory
+                return false;
```

Comment:
we need to bring back fs::canonical, to be sure the path is correct.

'cleo/../../../../windows/system32' is not a safe path.

Reply:
if "gta san andreas" is the game directory, then 

`cleo/../../gta san andreas/cleo` is safe

Reply:
if "gta san andreas" is the game directory, then

`../gta san andreas/cleo` is safe

Reply:
FilepathNormalize is taking care of collapsing "..\"

Reply:
We don't call it on `path` if workdir is empty

Reply:
FilepathNormalize updated:
cleo/../../a/b > ../../a/b
/../../a/b > ../../a/b
../../a/b > ../../a/b

Reply:
> We don't call it on `path` if workdir is empty
Comment requires input argument to be normalized already. Function is used inside read opcode path argument macro, which performs normalization first.

Reply:
> cleo/../../a/b > ../../a/b

why? what happened to cleo folder?

Reply:
> `cleo/../../gta san andreas/cleo` is safe

That can be resolved only in context of full path. We already stripped some other functions of returning absolute paths in case of game root. Now it is impossible to build absolute path.
We again come to realization we need no ambiguities in paths processing and have special cases only when asking ModLoader for files.

Reply:
should be ../a/b

Reply:
I'm tired of this ModLoader argument.

> Now it is impossible to build absolute path.

It is possible with fs::canonical, but we removed it because of performance issues if called too often

Reply:
> should be ../a/b

Mistake in comment.
![image](https://github.com/user-attachments/assets/4bd0ff45-098b-4a3d-a1ef-1ba310ccc59a)

Reply:
From what I read [here](https://en.cppreference.com/w/cpp/filesystem/canonical) both canonical and weakly_canonical work with relative paths (converting them to absolute).